webdriver-spec.html

```diff
@@ -5316,106 +5316,110 @@
-      For the former case, the current tick being executed must wait
-      for the longest pause to complete.
-      For example, in this diagram:
-
-     <p><img src=graphics/note2actions.svg alt="">
-
-     <p>The remote end will dispatch the <code><a>pointerDown</a></code> action in the first tick.
-      In the second tick, since source 1 declares a <code><a>pause</a></code> of 5 seconds,
-      the remote end will dispatch the <code><a>pointerUp</a></code> event for source 2,
-      and will wait 5 seconds before moving on to executing the third tick.
-
-     <p>In the event that one tick contains multiple <code><a>pause</a></code> durations,
-      the remote end will wait the maximum duration
-      before moving on to executing the next tick.
-
-     <p>As noted before, <code><a>pause</a></code> can be used to signify inaction during a tick.
-      If <code><a>pause</a></code> is declared without a time period,
-      then the input source will not have any actions executed in the containing tick.
-      As an example:
-
-     <p><img src=graphics/note3actions.svg alt="">
-
-     <p>During tick 2, source 1 will have its pointerMove action dispatched,
-      while source 2 will do nothing.
-  </aside>
-
-  <p>
-
-  </section> <!-- /Introduction -->
+<section>
+<h2>Actions</h2>
+
+<p class=issue>Expand these examples
+
+<p>The Actions API provides a low-level interface
+ for providing virtualised device input to the web browser.
+ Conecputally, the actions API divides time into a series of <a>ticks</a>.
+ The local end sends a series of actions
+ which correspond to the change in state, if any,
+ of each input device during each tick.
+ For example, pressing a key
+ is represented by an action chain
+ consisting of a single key input device and two ticks,
+ the first containing a <a>keyDown</a> action,
+ and the second a <a>keyUp</a> action,
+ whereas a pinch-zoom input is represented
+ by an action chain consisting of a three ticks
+ and two pointer input devices of type touch,
+ each performing a sequence of actions <a>pointerDown</a>,
+ followed by <a>pointerMove</a>, and then <a>pointerUp</a>.
+
+<p class=issue>Move this note:
+
+<p class=note>An <a>activation trigger</a>
+ generated by the WebDriver API user
+ need to be indistinguishable from those generated
+ by a real user interacting with the browser.
+ In particular, the dispatched events will have
+ the <a>isTrusted</a> attribute set to true.
+ The most robust way to dispatch these events
+ is by creating them in the browser implementation itself.
+ Sending OS-specific input messages to the browser's window
+ has the disadvantage that the browser being automated
+ may not be properly isolated from a user
+ accidentally modifying input source state;
+ use of an OS-level accessibility API has the disadvantage
+ that the browser's window must be focused,
+ and as a result, multiple WebDriver instances cannot run in parallel.
```

Comment:
Might be worth calling out that the advantage of a OS-level accessibility API is that it guarantees inputs correctly mirror user input, and allows interaction with the host OS if necessary. Though using it is a terrible idea from a machine utilisation perspective.